core/src/main/scala/org/apache/spark/util/SignalUtils.scala

```diff
@@ -31,14 +31,51 @@ import org.apache.spark.internal.Logging
 /**
  * Contains utilities for working with posix signals.
  */
-private[spark] object Signaling extends Logging {
+private[spark] object SignalUtils extends Logging {
+
+  private var registered = false
+
+  /** Register a signal handler to log signals on UNIX-like systems. */
+  def registerLogger(log: Logger): Unit = synchronized {
```

Comment:
that's true :) I think it's just a matter of preference and what you consider is distinct functionality (I actually started by removing logging but then re-amending it after second thought). I agree that both classes provide only utilities so it's really a minor issue.
My only concern is the use of the `registered` variable (see below)